var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -46,6 +46,13 @@ class Petsc(Package):
     version('3.5.1', 'a557e029711ebf425544e117ffa44d8f')
     version('3.4.4', '7edbc68aa6d8d6a3295dd5f6c2f6979d')
 
+    #instead of needing to manually provide specific git tags in the package.py for each package could there 
+    #   be a way to describe the git tag/hash in the spack spec such a
+    #   @git:tag_or_commit_hash using maybe
+    #version('*', git='https://bitbucket.org/petsc/petsc.git')
+    #   then users could do petsc@master or petsc@v3.7.1 or petsc@11bc23.. ?
```

Comment:
You can provide tag versions like this:

```python
version('1.5', git='<URL>', tag='<tag>')
```

Same for `commit` and `branch` (see [fetching from VCS repositories](http://spack.readthedocs.io/en/latest/packaging_guide.html#fetching-from-vcs-repositories)).

Making it automatic would be nice, but keep in mind that Spack is supposed to work behind an air gap -- so the package can't expect a connection to the git server.  The versions have to be enumerated here so the Spack can know what to fetch if it's pulling from an https or filesystem mirror.